spec/services/course/duplication/course_duplication_service_spec.rb

```diff
@@ -498,6 +498,33 @@
           expect(all_new_comment_emails_disabled).to be(true)
         end
       end
+
+      context 'when sidebar settings have multiple assessment categories' do
+        let!(:categories) { create_list(:course_assessment_category, 2, course: course) }
+
+        before do
+          # Assign sidebar settings for original course by setting the weight to the old
+          # category ID.
+          course.assessment_categories.each do |category|
+            course.settings(:sidebar).settings("assessments_#{category.id}").weight = category.id
+          end
+        end
+
+        it 'updates the sidebar setting keys with the new assessment category IDs' do
+          weights = course.assessment_categories.map(&:id)
+          # Check that sidebar keys with the new assessment category IDs are present and have
+          # values in the original set of weights.
+          new_course.assessment_categories.each do |category|
+            expect(weights).
+              to include(course.settings(:sidebar).settings("assessments_#{category.id}").weight)
+          end
```

Comment:
By right you need to check against `new_course.settings(:sidebar).settings("assessments_#{category.id}").weight)`, and not `course.settings...`?

Can I also suggest the following, cause it took me a while to read through and understand this, and I thought this suggestion would be more readable (without the need for comments):
```ruby
old_course_weights = course.assessment_categories.map(&:id)
new_course_weights = new_course.assessment_categories.map do |c| 
  new_course.settings(:sidebar).settings("assessments_#{c.id}").weight
end
expect(new_course_weights).to contain_exactly(*old_course_weights)
```

Reply:
Yes should be `new_course.settings...`, not sure why the test still passed.

Ok, will change to that implementation.

Reply:
eh, I think should be `old_course`, because we are checking whether the value is duplicated.